src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1734,6 +1734,16 @@ public function isXmlHttpRequest()
         return 'XMLHttpRequest' == $this->headers->get('X-Requested-With');
     }
 
+    /**
+     * Returns true if the request is a CORS-preflight request.
+     *
+     * @see https://fetch.spec.whatwg.org/#http-requests
+     */
+    public function isCorsPreflightRequest(): bool
+    {
+        return $this->isMethod('OPTIONS') && $this->headers->has('Access-Control-Request-Method');
```

Comment:
would it make sense to also check for `Origin` and `Access-Control-Request-Headers`?